homeassistant/helpers/entity_platform.py

```diff
@@ -844,13 +846,20 @@ async def _update_entity_states(self, now: datetime) -> None:
             return
 
         async with self._process_updates:
-            tasks: list[Coroutine[Any, Any, None]] = []
-            for entity in self.entities.values():
-                if not entity.should_poll:
-                    continue
-                tasks.append(entity.async_update_ha_state(True))
+            if self._update_in_parallel or len(self.entities) <= 1:
+                # If we know are going to update sequentially, we want to update
+                # to avoid scheduling the coroutines as tasks that will we know
+                # are going to wait on the semaphore lock.
+                for entity in list(self.entities.values()):
+                    if entity.should_poll and entity.hass:
```

Comment:
Checking `entity.hass` looks like new code not related to the PR subject.

Reply:
The entity can be removed from hass while we are awaiting the previous entity in the list.  Will add a comment

Reply:
https://github.com/home-assistant/core/pull/93693